### DISCUSSION TITLE
How can I host a zART project? React Native

### DISCUSSION THREAD

**Discussion by Kerim-Willem** [trpc/trpc#2566] [Q&A] [answered]
Hi there community! 

My question is: How can I host a zART project? 

I want to start using trpc, and create a nextjs/react-native app while using trpc. But I'm quite stuck on figuring out how to actually host this.  

What I don't understand about this package, is how we separate hosting Nextjs and the trpc server on Vercel, from hosting React Native and having trcp connected at the same. 

How it seems to me, is that you bundle React Native into a package, which you import to the App Store and Google Play Store. But do you import the entire app (including the Nextjs/API part) into the App store and Google Play Store? Or do you just import the React Native part? If you just import the React Native part, then how do you connect with trpc?

**Comment by mariandaniel24** [comment, marked as answer]
> What I don't understand about this package, is how we separate hosting Nextjs and the trpc server on Vercel, from hosting React Native and having trcp connected at the same.

tRPC is just a package that runs on top of existing web libraries (express/fastify/vanilla node http server), it just adds type safety at compile time. At runtime, there is no TS typechecking involved, that means you can easily deploy tRPC anywhere, as it's on "module", any requests  or data-exchanges are done through regular HTTP calls.

> But do you import the entire app (including the Nextjs/API part) into the App store and Google Play Store?

You don't have to.

>  Or do you just import the React Native part? If you just import the React Native part, then how do you connect with trpc?

Correct.  There are two parts that you need to "connect":
1. The tRPC router type definitions ( this is done through the `import type { AppRouter } from 'wherever-your-router-is-located'`.)
 Keep in mind that these types are only imported and checked at compile time, absolutely zero server code will end up in your `react-native` / client
2. Communication between client and server is done through an HTTP endpoint, in the case of a next.js app, the endpoint path will be mapped by next.js ( `/next-js-app/src/pages/api/trpc/[trpc].ts` -> `http://localhost:3000/api/trpc`).
The main reason tRPC lives inside the next.js folder structure is because next.js already allows you to create API endpoints, so it wouldn't make much sense to have a separate `node` server that does the same thing (although you can definitely do it technically)

**Reply by Kerim-Willem** [reply]
Thank you very much for this explanation! One thing that I don't completely understand yet is: 

> The tRPC router type definitions ( this is done through the import type { AppRouter } from 'wherever-your-router-is-located'.)

Does this mean I have to expose the types I create from the trpc server as a npm package?

**Reply by mariandaniel24** [reply]
You don't have to publish/expose types an NPM package, the path to the server's `AppRouter` should be a file path in your project, so something like this would be valid:
`import type { AppRouter } from '../../../../server/router.ts';`
I believe the zART stack uses yarn workspaces to avoid `../../` hell, but it's gonna work similarly, just a different import path, ultimately it's going to map to the same tRPC router in your project.

If you're already using zART, this will all be set up for you out of the box, I hope this just helps you understand how this is wired up under the hood.

**Reply by mariandaniel24** [reply]
When deploying your react-native app, make sure that [this](https://github.com/trpc/zart/blob/main/apps/expo/App.tsx#L14) path points to your production Vercel URL.

**Comment by Kerim-Willem** [comment]
Aaah I finally understand now! Thank you so much!
